src/migrations/user-store/5.0.3-beta.1.ts

```diff
@@ -50,6 +52,13 @@ export default {
           continue;
         }
 
+        const relativeToExtensionData = path.relative(cluster.kubeConfigPath, extensionDataDir);
+
+        if (relativeToExtensionData === "" || relativeToExtensionData.match(/^(\.\.)([\\\/]\.\.)*$/)) {
+          migrationLog(`Skipping ${cluster.id} because kubeConfigPath is placed under an extension_data folder`);
+          continue;
+        }
```

Comment:
without regex: https://stackoverflow.com/a/45242825

Reply:
I agree that solution is nice but it fails if the folder or file name starts with `..`.

Reply:
I have done some more testing and have figured out a more robust way of checking this. Will update PR with tests.

Reply:
What path starts with `..`? `extensionDataDir` is an absolute path, right?

Reply:
A folder or file.

Reply:
when will you encounter a file or folder that starts with `..`? `path.resolve()` always returns an absolute path, and that's what you're starting with. `extensionDataDir` and `cluster.kubeConfigPath` are absolute. I don't see what's wrong with https://stackoverflow.com/a/45242825

Reply:
I mean the name of the folder is `..blah` so the path could be `/foo/bar/..blah`

Reply:
oh, ugh.